### PR TITLE
Allow clients to set request headers

### PR DESCRIPTION
**Pull request by yunapotamus**

### CODE REVIEW

File: Sources/PromotedFirebaseRemoteConfig/ClientConfig+PromotedFirebaseRemoteConfig.swift [hunk 1]
```diff
@@ -200,13 +203,10 @@ extension String {
...
   func toCamelCase() -> String {
-    var first = true
     return split(separator: "_")
-      .map { value in
-        let result = first ? value :
-          value.prefix(1).uppercased() + value.dropFirst()
-        first = false
-        return String(result)
+      .enumerated()
+      .map { (index, str) in
```
Comment (danbosnichill) [inline Sources/PromotedFirebaseRemoteConfig/ClientConfig+PromotedFirebaseRemoteConfig.swift:208]: Interesting.  This looks better.  I'm surprised this doesn't break things.

**Review by danbosnichill — APPROVED** [review]

File: Sources/PromotedFirebaseRemoteConfig/ClientConfig+PromotedFirebaseRemoteConfig.swift (hunk 1 above)
Comment (yunapotamus) [reply Sources/PromotedFirebaseRemoteConfig/ClientConfig+PromotedFirebaseRemoteConfig.swift:208]: Tests still pass. I added a new test to cover an edge case with double underscores.
